src/extension/features/budget/display-total-monthly-goals/index.js

```diff
@@ -17,71 +22,156 @@ export class DisplayTotalMonthlyGoals extends Feature {
       return;
     }
 
+    console.log(`Category Name: ${category.name}`);
+    console.log(`Category Goal: ${category.goalTarget}`);
+    console.log(`Category Type: ${category.goalType}`);
+
     return {
-      monthlyGoalAmount: parseInt(category.goalTarget || 0, 10),
-      isChecked: category.get('isChecked'),
+      name: category.name,
+      type: category.goalType,
+      goal: parseInt(category.goalTarget || 0, 10),
     };
   }
 
-  calculateMonthlyGoals() {
-    const categoryGoals = {
-      total: 0,
-      checkedTotal: 0,
-      checkedCount: 0,
-    };
+  calculateTotalAssigned() {
+    // Get current month and year
+    const currentBudgetDate = getCurrentBudgetDate();
+    const currentYear = parseInt(currentBudgetDate.year);
+    const currentMonth = parseInt(currentBudgetDate.month);
+
+    // Get all budget calculations from YNAB
+    const allBudgetCalculations = getEntityManager().getAllMonthlyBudgetCalculations();
+
+    // Find current month budget calculations
+    const budgetedCalculation = allBudgetCalculations.filter((budgetItem) => {
+      const budgetItemDate = budgetItem.monthlyBudgetId.split('/')[1].split('-');
+      return (
+        currentYear === parseInt(budgetItemDate[0]) && currentMonth === parseInt(budgetItemDate[1])
+      );
+    })[0];
+
+    console.log(budgetedCalculation);
+
+    const budgeted = budgetedCalculation?.budgeted || 0;
+    const income = budgetedCalculation?.immediateIncome || 0;
+    const spent = budgetedCalculation?.cashOutflows || 0;
+
+    return { income, budgeted, spent };
+  }
+
+  calculateTotalGoals() {
+    var savingsGoals = 0;
+    var spendingGoals = 0;
 
     $('.budget-table-row.is-sub-category').each((_, element) => {
-      const categoryGoal = this.extractCategoryGoalInformation(element);
-      if (!categoryGoal) {
+      const goalData = this.extractCategoryGoalInformation(element);
+      if (!goalData) {
         return;
       }
 
-      categoryGoals.total += categoryGoal.monthlyGoalAmount;
-      if (categoryGoal.isChecked) {
-        categoryGoals.checkedTotal += categoryGoal.monthlyGoalAmount;
-        categoryGoals.checkedCount++;
-      }
+      if (goalData.type === 'MF') savingsGoals += goalData.goal;
```

Comment:
I should really add a lint rule for this, can we please use brackets for this?

also, I don't think we should be removing the `isChecked` check. seeing my totally monthly goals for selected categories is actually kinda nice

Reply:
Hey @joshmadewell thanks for the input here.  I definitely agree about the isChecked functionality being left in.  I don't think it makes much sense to have checked show in the totals for my use case though,  so will move what I'm working on into its own feature.

As far as the brackets you were mentioning, I can't think of what you might be referring to off the top of my head.  Could you expand a bit on that?